clean-abap/CleanABAP.md

```diff
@@ -2106,6 +2107,42 @@ CLASS cl_my_class IMPLEMENTATION.
   ...
 ```
 
+#### Don't access types through instance variables
+
+> [Clean ABAP](#clean-abap) > [Content](#content) > [Methods](#methods) > [Calls](#calls) > [This section](#dont-access-types-through-instance-variables)
+
+When using a data type that is defined in a class or interface, access the type definition via the class/interface and not via an instance of the class/interface. 
+
+```ABAP
+CLASS lcl DEFINITION.
+  PUBLIC SECTION.
+    TYPES foo TYPE i.
+ENDCLASS.
+CLASS lcl IMPLEMENTATION.
+ENDCLASS.
+
+INTERFACE lif.
+  TYPES blah TYPE lcl=>foo.  
+ENDINTERFACE.
+```
+
+Using the instance for the data type would be confusing, as it suggests that the type is instance specific. 
```

Comment:
I generally don't agree with the inclusion of this rule, but I'd like to understand your point of view.

How is using the instance for the data type confusing? Types are never instance-specific.

Reply:
The idea was pretty exactly that: Types can never be instance-specific, so why should we make it look like the type belonged to the instance?